src/store/index.js

```diff
@@ -1,31 +1,40 @@
 import Vue from 'vue'
 import Vuex from 'vuex'
 import PNPStore from './pnp.js'
-import { INITIALIZE_PNP } from './action-types.js'
-import { UPDATE_AVAILABLE } from './mutation-types'
+import { INITIALIZE_PNP, FETCH_EDGE_DEVICE_DETAILS } from './action-types.js'
+import { UPDATE_AVAILABLE, EDGE_DEVICE_DETAILS } from './mutation-types'
 
 Vue.use(Vuex)
 
-const store = new Vuex.Store({
+const applicationStore = {
   state: {
     updateToBeInstalled: undefined,
     version: require('@/../package.json').version
   },
   mutations: {
     [UPDATE_AVAILABLE] (state, updateToBeInstalled) {
       state.updateToBeInstalled = updateToBeInstalled
+    },
+    [EDGE_DEVICE_DETAILS] (state, edgeDetails) {
+      state.version = edgeDetails.version
     }
   },
   actions: {
+    [FETCH_EDGE_DEVICE_DETAILS] ({ commit }, details) {
+      console.log('I AM CALLED')
+      commit(EDGE_DEVICE_DETAILS, details)
+    }
   },
   modules: {
     pnp: PNPStore
   }
-})
+}
+
+const store = new Vuex.Store(applicationStore)
```

Comment:
@vickywane can you please elaborate on the semantic use of applixationStore in addition to the Vuex provided store structure? Why do we need to export both in this file?

Reply:
@ivelin i exported the applicationStore so i can isolate and test it when testing components connected it. Being the default store, i couldnt write tests for it like the `pnp` store that’s exported and used as a Vuex module. 

See an example usage [here](https://github.com/ambianic/ambianic-ui/pull/582/files#diff-a0e2b31817857d0241e7b5084937ab2ba1bfe4c411405ec24d6ba64a392b3e9f)

Reply:
@vickywane Let's move the new code to a separate module. Let's put EDGE_DEVICE_DETAILS and FETCH_EDGE_DEVICE_DETAILS in an edgeDevice module in the Vuex store instead of the default top level store.